commands/ldap_e2e_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package commands
+
+import (
+	"os"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/api4"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/utils/testutils"
+	"github.com/spf13/cobra"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func configForLdap(th *api4.TestHelper) {
+	ldapHost := os.Getenv("CI_LDAP_HOST")
+	if ldapHost == "" {
+		ldapHost = testutils.GetInterface(*th.App.Config().LdapSettings.LdapPort)
+	}
+
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.ServiceSettings.EnableMultifactorAuthentication = true
+		*cfg.LdapSettings.Enable = true
+		*cfg.LdapSettings.EnableSync = true
+		*cfg.LdapSettings.LdapServer = "dockerhost"
+		*cfg.LdapSettings.BaseDN = "dc=mm,dc=test,dc=com"
+		*cfg.LdapSettings.LdapServer = ldapHost
+		*cfg.LdapSettings.BindUsername = "cn=admin,dc=mm,dc=test,dc=com"
+		*cfg.LdapSettings.BindPassword = "mostest"
+		*cfg.LdapSettings.FirstNameAttribute = "cn"
+		*cfg.LdapSettings.LastNameAttribute = "sn"
+		*cfg.LdapSettings.NicknameAttribute = "cn"
+		*cfg.LdapSettings.EmailAttribute = "mail"
+		*cfg.LdapSettings.UsernameAttribute = "uid"
+		*cfg.LdapSettings.IdAttribute = "cn"
+		*cfg.LdapSettings.LoginIdAttribute = "uid"
+		*cfg.LdapSettings.SkipCertificateVerification = true
+		*cfg.LdapSettings.GroupFilter = ""
+		*cfg.LdapSettings.GroupDisplayNameAttribute = "cN"
+		*cfg.LdapSettings.GroupIdAttribute = "entRyUuId"
+		*cfg.LdapSettings.MaxPageSize = 0
+	})
+
+	th.App.Srv().SetLicense(model.NewTestLicense("ldap"))
+}
+
+func (s *MmctlE2ETestSuite) TestLdapSyncCmd() {
+	s.SetupEnterpriseTestHelper().InitBasic()
+	configForLdap(s.th)
+
+	s.Run("Should not allow regular user to sync LDAP groups", func() {
+		printer.Clean()
+
+		err := ldapSyncCmdF(s.th.Client, &cobra.Command{}, nil)
+		s.Require().Error(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.RunForSystemAdminAndLocal("Should sync LDAP groups", func(c client.Client) {
+		printer.Clean()
+
+		jobs, appErr := s.th.App.GetJobsByTypePage(model.JOB_TYPE_LDAP_SYNC, 0, 100)
+		s.Require().Nil(appErr)
+		initialNumJobs := len(jobs)
+
+		err := ldapSyncCmdF(c, &cobra.Command{}, nil)
+		s.Require().NoError(err)
+
+		s.Require().NotEmpty(printer.GetLines())
+		s.Require().Equal(printer.GetLines()[0], map[string]interface{}{"status": "ok"})
+		s.Require().Len(printer.GetErrorLines(), 0)
+
+		// we need to wait a bit for job creation
+		time.Sleep(time.Second)
+
+		jobs, appErr = s.th.App.GetJobsByTypePage(model.JOB_TYPE_LDAP_SYNC, 0, 100)
+		s.Require().Nil(appErr)
+		s.Require().NotEmpty(jobs)
+		s.Assert().Greater(len(jobs), initialNumJobs)
```

Comment:
does it make sense to be more precise and test for `len(jobs) + 1`?

Reply:
Makes sense 👍